json-rpc/src/views.rs

```diff
@@ -40,3 +40,52 @@ impl From<&[u8]> for BytesView {
         Self(hex::encode(bytes))
     }
 }
+
+#[derive(Serialize)]
+pub struct TransactionView {
+    pub version: u64,
+    pub transaction: TransactionData,
+}
+
+#[derive(Serialize)]
+pub struct TransactionData {
+    pub blockmetadata: Option<BlockMetadataTransactionView>,
+    pub writeset: Option<WriteSetTransactionView>,
+    pub user: Option<UserTransactionView>,
+}
+
+#[derive(Serialize)]
```

Comment:
can we group these (`BlockMetadataTransactionView`, `WriteSetTransactionView`, `UserTransactionView`) into enum values? that would help with organizing the objects conceptually, especially since there is also `TransactionView` and `BlockMetadata` above in this file

Reply:
also do we have to add `BlockMetadataTransactionView` and `WriteSetTransactionView` in this PR? Because even in the tests there isn't much to check for them anyway. My personal preference is to add and test them robustly once specs are complete/clarified - but not a huge deal I guess.

Reply:
I wasn’t sure how enum will get serialized/deserialized .  I found https://serde.rs/enum-representations.html  which style do you prefer?  I chose struct because it was kind of obvious what the json would look like. If we choose Untagged version, the user is impossible to know the type, if we choose tagged version it allows for more dynamicism, but we will possibly only ever have 3 it seems unnecessary complex to the user. That’s why I chose struct 

About blockmeatdata and Writesets These transactions always exists and actually consists majority of the real transactions api returned. If we don’t have placeholder for them we will have to return empty objects which would be bad.

Reply:
I like internal representation with enums. @thefallentree: which one do you like. My preference is around "Internally tagged". I guess that's what I used to the most

Reply:
+1 for "internally tagged"

Reply:
done.